client.go

```diff
@@ -292,6 +297,42 @@ func (client *client) RefreshMetadata(topics ...string) error {
 	return client.tryRefreshMetadata(topics, client.conf.Metadata.Retry.Max)
 }
 
+func (client *client) GetOffsetRange(topic string, partitionID int32) (int64, int64, error) {
```

Comment:
it only happens once on startup, doesn't seem like a big deal (premature optimization and all that)